lightningd/options.c

```diff
@@ -1049,6 +1051,14 @@ static char *opt_set_db_upgrade(const char *arg, struct lightningd *ld)
 	return opt_set_bool_arg(arg, ld->db_upgrade_ok);
 }
 
+static char *opt_disable_ip_discovery(struct lightningd *ld)
+{
+	log_broken(ld->log, "--disable-ip-discovery has been deprecated, use --ip-discovery");
+	if (!deprecated_apis)
+		return "--disable-ip-discovery has been deprecated, use --ip-discovery";
+	return opt_set_bool(&ld->config.disable_ip_discovery);
```

Comment:
Can't you just set the autobool to false instead?

Reply:
Yep, thats better. I just log_broken and set the new config value to false and `return NULL` to proceed.